src/coreclr/jit/gentree.h

```diff
@@ -5303,6 +5304,13 @@ struct GenTreeCall final : public GenTree
         return (gtCallMoreFlags & GTF_CALL_M_PINVOKE) != 0;
     }
 
+    // Returns true if this call represents a volatile operation.
+    // Typically, memset/memcpy helper calls for volatile. cpblk/initblk
+    bool IsVolatile() const
+    {
+        return (gtCallMoreFlags & GTF_CALL_M_VOLATILE) != 0;
+    }
```

Comment:
Can we add another node instead? I think we should stop trying to overload `GT_CALL` like this...

Reply:
> Can we add another node instead? I think we should stop trying to overload `GT_CALL` like this...

What do you mean by another node?
This flag can be made precisely MEMSET/MEMCPY helper call specific so its value can be re-used

Reply:
`GT_MEMORYBARRIER` already exists -- can it be reused?

Reply:
I can also just insert `GT_MEMORYBARRIER` in Lower

Reply:
> `GT_MEMORYBARRIER` already exists -- can it be reused?

Yeah I had it in mind, just assumed that if I also remove GT_STORE_DYN_BLK early in importer - would I want to carry GT_MEMORYBARRIER all the way down?

From the other hand, I presume volatile. cpblk/initblk are extremely rare

Reply:
Addressed